certbot/compat/filesystem.py

```diff
@@ -36,6 +43,67 @@ def chmod(file_path, mode):
         _apply_win_mode(file_path, mode)
 
 
+# Funtion os.makedirs will temporarily monkeypatch os.mkdir to replace it by filesystem.mkdir.
+# However, filesystem.mkdir also invokes os.mkdir. To avoid a looped reference, we take now a
+# reference of the original os.mkdir, to use it in filesystem.mkdir.
+_os_mkdir = os.mkdir
+
+
+def makedirs(file_path, mode=0o777, exists_ok=False):
+    # type: (str, int, bool) -> None
+    """
+    Rewrite of original os.makedirs function, that will ensure on Windows that given mode
+    is correctly applied.
+    :param str file_path: The file path to open
+    :param int mode: POSIX mode to apply on leaf directory when created, Python defaults
+                     will be applied if ``None``
+    :param bool exists_ok: If set to ``True``, do not raise exception if leaf directory
+                           already exists.
+    """
+    # As we know that os.mkdir is called internally by os.makedirs, we will swap the function in
+    # os module for the time of makedirs execution.
+    orig_mkdir_fn = os.mkdir
+    try:
+        os.mkdir = mkdir  # type: ignore
+        try:
+            os.makedirs(file_path, mode)
+        except (IOError, OSError) as err:
+            # In case of exists_ok is True, and exception was about the path to already exist,
+            # we ignore the exception.
+            if not exists_ok or err.errno != errno.EEXIST:
+                raise
+    finally:
+        os.mkdir = orig_mkdir_fn
+
+
+def mkdir(file_path, mode=0o777):
+    # type: (str, int) -> None
+    """
+    Rewrite of original os.mkdir function, that will ensure on Windows that given mode
+    is correctly applied.
+    :param str file_path: The file path to open
+    :param int mode: POSIX mode to apply on directory when created, Python defaults
+                     will be applied if ``None``
+    """
+    if POSIX_MODE:
+        _os_mkdir(file_path, mode)
+    else:
```

Comment:
nit: if we return instead of `else`ing, everything else can be one less indent...

Reply:
Done.